Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +20,150 @@
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                if (localNetworkNodeInfoList != null) {
+                    networkNodeInfoList.addAll(localNetworkNodeInfoList);
+                }
+            } catch (QuadrigaStorageException e) {
+                throw new QuadrigaStorageException("Database Error while getting network top nodes for a network with ID: " +
+                        network.getNetworkId(), e);
+            }
+        }
+
+        ITransformedNetwork networkJSon = transformer.transformNetwork(networkNodeInfoList);
+        return networkJSon;
+    }
+
+    @Override
+    public ITransformedNetwork searchTransformedNetworkOfProject(String projectId, String conceptId)
+            throws QuadrigaStorageException {
+        List<INetwork> networkList = getNetworkList(projectId);
+        if (networkList == null) {
+            return null;
+        }
+
+        Map<String, Node> searchNodes = new HashMap<String, Node>();
+        List<Link> searchLinks = new ArrayList<Link>();
+        for (INetwork network : networkList) {
+            try {
+                List<INetworkNodeInfo> localNetworkNodeInfoList =
+                        networkManager.getNetworkTopNodes(network.getNetworkId());
+                // check if this network has the node with given concept id
+                if (localNetworkNodeInfoList != null) {
+                    ITransformedNetwork transformedNetwork =
+                            transformer.transformNetwork(localNetworkNodeInfoList);
+                    // search for concept id
+                    Node searchNode = null;
+                    for (Node node: transformedNetwork.getNodes().values()) {
+                        if (node.getConceptId().equals(conceptId)) {
+                            searchNode = node;
+                            break;
+                        }
+                    }
+                    // if node does not exist continue
+                    if (searchNode == null) {
+                        continue;
+                    }
+
+                    Set<String> searchStatements = new HashSet<String>(searchNode.getStatementIds());
+                    // String statementId = searchNode.getStatementIds().get(0);
+                    Set<Node> associatedNodes = new HashSet<Node>();
+                    for (Map.Entry<String, Node> entry: transformedNetwork.getNodes().entrySet()) {
+                        // check if first statementId is in all the nodes
+                        Node n = entry.getValue();
+                        for (String statementId: n.getStatementIds()) {
+                            if (searchStatements.contains(statementId)) {
+                                searchNodes.put(entry.getKey(), n);
+                                associatedNodes.add(n);
+                            }
+                        }
+                    }
```

Comment:
The problem is TransformedNetwork class has not been setup properly. Even if I had a map with statement id and list of nodes, I still have to search in all the entries in the map for a given concept id.

A lot of code has to be changed in order to fix the TransformedNetwork class. I have come up with a different solution which is a lot similar to yours.

1. Get the combined network of the project
2. Add all the nodes which has a given concept id in a list
3. I will reduce the nodes to map<string, set<Node>> where string is the statement id and node is the one which is included in that statement.
4. Now with this I can get all the statements which contains the concept id
5. I will remove other nodes and links from the transformed network.
6. Please refer to this pull request - https://github.com/diging/quadriga/pull/143